python/ray/serve/_private/deployment_state.py

```diff
@@ -1311,10 +1311,7 @@ def get_checkpoint_data(self) -> DeploymentTargetState:
     def recover_target_state_from_checkpoint(
         self, target_state_checkpoint: DeploymentTargetState
     ):
-        logger.info(
-            f"Recovering target state for deployment {self.deployment_name} in "
-            f"application {self.app_name} from checkpoint."
-        )
+        logger.info(f"Recovering target state for {repr(self._id)} from checkpoint.")
```

Comment:
Non-blocker: I think by default if `__str__()` is not defined, `__repr__()` will be used automatically. No need to explicitly calling repr(), but no harm to be explicit either :)

Reply:
If that's the case should we remove `__str__()` for `DeploymentID`? I believe we added that while trying to maintain backwards compatibility with metrics a while ago, but the code migration is already done so that's not necessary anymore.

Reply:
oh nice if it's possible to remove the existing `app_deployment` `__str__` method then I can just implement `__str__` and not explicitly call `__repr__` here

Reply:
updated